canopy_utils_mod.F90

```diff
@@ -70,5 +70,23 @@ function CalcPAI(fch, ffrac)
         CalcPAI=( (fch*(ffrac/3.0_rk)*10.6955_rk) / (2.0_rk * pi) ) * ffrac !Massman PAI calculation (Eq. 19)
 
     end function
+    !--------------------------------------------------------------------------------------
+
+    function CalcFlameH(frp, dx)
+        !! Approximates the Flame Height as a function of FRP intensity and grid cell length (dx)
+        !! forest fraction (Based on Byram 1959).
+
+        !!  Byram, GM (1959). Combustion of Forest Fuels. In Forest Fire: Control and Use.
+        !!  (Ed. KP David) pp. 61-89.  McGraw Hill, New York, NY
+
+        !! Assume Flame Length = Flame Height under calm winds
+
+        real(rk), intent(in)  :: frp          !! Input Grid cell Fire Radiative Power (MW/cell)
+        real(rk), intent(in)  :: dx           !! Input Grid cell length (m)
+        real(rk)              :: CalcFlameH   !! Calculated Plant area index (PAI)
+
+        CalcFlameH=0.0775*((frp*1000.0)/dx)**0.46  !Byram flameh calculation as function of FRP
```

Comment:
It is true, and I modify the dlon calc to take absolute value now first,  so that dx is never negative.    However, dx = 0 can be valid if we are running this for a single point, i.e., nlon = 1.  I have this condition covered later, which sets back to the user flameh_set value in the namelist, as if there is no fire/FRP intensity over model grid distance flameh cannot be calculated.
...
...
if (nlon .eq. 1 .or. dx .eq. 0) then      !single lon point -- reset to user value
                    flameh = flameh_set